docs/contributing/testing-guidelines/unit-testing.md

```diff
@@ -0,0 +1,168 @@
+CMake-based package and to ensure that JUnit-compatible result files are generated. It currently
+supports a few different testing frameworks like `pytest`, `gtest`, and `gmock`.
+
+In order to prevent tests running in parallel from interfering with each other when publishing and subscribing to ROS topics,
+it is recommended to use commands from [`ament_cmake_ros`](https://github.com/ros2/ament_cmake_ros/tree/master/ament_cmake_ros/cmake) to run tests in isolation.
+
+See below for an example of using `ament_add_ros_isolated_gtest` with `colcon test`.
+All other tests follow a similar pattern.
+
+## Create a unit test with gtest
+
+In `my_cool_pkg/test`, create the `gtest` code file `test_my_cool_pkg.cpp`:
+
+```{cpp}
+#include "gtest/gtest.h"
+#include "my_cool_pkg/my_cool_pkg.hpp"
+TEST(TestMyCoolPkg, TestHello) {
+  EXPECT_EQ(my_cool_pkg::print_hello(), 0);
+}
+```
+
+For more examples of `gtest` features, see the
+[gtest repo](https://github.com/google/googletest).
+
+In `package.xml`, add the following line:
+
+```{xml}
+<test_depend>ament_cmake_ros</test_depend>
+```
+
+Next add an entry under `BUILD_TESTING` in the `CMakeLists.txt` to compile the test
+source files:
+
+```{cmake}
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+
+  ament_add_ros_isolated_gtest(${TEST_MY_COOL_PKG} test/test_my_cool_pkg.cpp)
+
+  target_link_libraries(${TEST_MY_COOL_PKG} ${PROJECT_NAME})
+...
+endif()
+```
+
+This automatically links the test with the default main function provided by `gtest`. The code under test is usually in a different CMake target (`${PROJECT_NAME}` in the example) and its shared object for linking need to be added.
+
+To register a new `gtest` item, wrap the test code with the macro `TEST ()`. `TEST ()`
+is a predefined macro that helps generate the final test code, and also registers
+a `gtest` item to be available for execution.
+The test case name should be in CamelCase, since gtest inserts an underscore between the fixture name and the class case name when creating the test executable.
+
+`gtest/gtest.h` also contains predefined macros of `gtest` like `ASSERT_TRUE(condition)`,
+`ASSERT_FALSE(condition)`, `ASSERT_EQ(val1,val2)`, `ASSERT_STREQ(str1,str2)`,
+`EXPECT_EQ()`, etc. `ASSERT_*` will abort the test if the condition is not
+satisfied, while `EXPECT_*` will mark the test as failed but continue to next test
+condition. More information about `gtest` can be found in the
+[gtest repo](https://github.com/google/googletest).
+
+In the demo `CMakeLists.txt`, `ament_add_ros_isolated_gtest` is a predefined macro in `ament_cmake_ros`
+that helps simplify adding `gtest` code. Details can be viewed in
+[ament_add_gtest.cmake](https://github.com/ros2/ament_cmake_ros/tree/master/ament_cmake_ros/cmake).
+
+## Build test
+
+By default, all necessary test files (`ELF`, `CTesttestfile.cmake`, etc.) are compiled by `colcon`:
```

Comment:
https://cmake.org/cmake/help/book/mastering-cmake/chapter/Testing%20With%20CMake%20and%20CTest.html

```suggestion
By default, all necessary test files (`ELF`, `CTestTestfile.cmake`, etc.) are compiled by `colcon`:
```

Reply:
I had to look this up because I never put a full stop after `etc` but apparently it is required. However, since the examples given are in parentheses in the middle of a sentence, I think the full stop can be omitted. This is purely from an aesthetic point of view however and not required.

```suggestion
By default, all necessary test files (`ELF`, `CTestTestfile.cmake`, etc) are compiled by `colcon`:
```

Reply:
Some examples from Google's style guide. It seems they add a full stop.

https://developers.google.com/style/word-list?hl=en
https://developers.google.com/style/api-reference-comments?hl=en
![image](https://user-images.githubusercontent.com/31987104/170275422-5942f50d-16be-4dd5-bfc2-34420f204a27.png)
![image](https://user-images.githubusercontent.com/31987104/170275550-4c63688e-0fea-4bf3-875a-8ae45f6937e5.png)